### PR TITLE
Track effective balance as number

### LINKED ISSUE #2981: Track effectiveBalance as small number
**Is your feature request related to a problem? Please describe.**

We should use number whenever possible as BigInt calculation is about x250 times slower as analysed by @dapplion 

**Describe the solution you'd like**

1. effectiveBalance is always a multiple of 1e9 and it's <= 32*1e9 so should be able to track 30, 31, 32 instead of BigInt(30 * 1e9), BigInt(31 * 1e9), BigInt(32 * 1e9). This would resolve issue #2747
2. In `epochProcess.ts`, totalActiveStake, prevEpochUnslashedStake and currEpochUnslashedTargetStake could also be tracked as number since they are sum of effectiveBalances
    + May need to tweak a little bit for some functions like `computeBaseRewardPerIncrement(), processEffectiveBalanceUpdates()`
    + Need to do an inventory for the usage of all of these variables but as far as I see it's all multiply/divide by `EFFECTIVE_BALANCE_INCREMENT` and this constant (1e9) is not likely to be changed

This should speed up `beforeProcessEpoch()` and some other calculations a lot since we don't have to deal with BigInt() there

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

+ BigInt is not necessary for effective balance
+ This should avoid a lot of params being configured as BigInt
+ forkchoice `computeDeltas()` wants to use number for its calculation instead of BigInts

waiting for https://github.com/ChainSafe/lodestar/pull/2973

**Description**

+ Change ssz type of Validator to use Number64 for `effectiveBalance`
+ In `beforeEpochProcess()`, we track total balance as multiple of `EFFECTIVE_BALANCE_INCREMENT` then when processing rewards and penalties we divide them to `EFFECTIVE_BALANCE_INCREMENT`, so we should track them "by increment" initially, in eth. It means we can use number there too.

Closes #2981

### CODE REVIEW

File: packages/lodestar/test/utils/state.ts [hunk 1]
```diff
@@ -83,7 +83,8 @@ export function generateState(
     eth1DataVotes: ([] as phase0.Eth1Data[]) as List<phase0.Eth1Data>,
     eth1DepositIndex: 0,
     validators: validators as List<phase0.Validator>,
-    balances: Array.from({length: 4}, () => MAX_EFFECTIVE_BALANCE) as List<Gwei>,
+    // TODO: fix once effective balance becomes number
+    balances: Array.from({length: 4}, () => Number(MAX_EFFECTIVE_BALANCE)) as List<number>,
```
Comment (dapplion) [inline packages/lodestar/test/utils/state.ts]: TODO

File: packages/types/src/primitive/sszTypes.ts [hunk 2]
```diff
@@ -17,7 +18,7 @@ export const Bytes96 = new ByteVectorType({length: 96});
 export const Uint8 = byteType;
 export const Uint16 = new NumberUintType({byteLength: 2});
 export const Uint32 = number32Type;
-export const Number64 = new NumberUintType({byteLength: 8});
+export const Number64 = new Number64UintType();
```
Comment (dapplion) [inline packages/types/src/primitive/sszTypes.ts]: Should we rename to `Number64Uint` to prevent issues in the future if we need a true `Number64`?

Comment (twoeths) [reply packages/types/src/primitive/sszTypes.ts]: yeah, maybe Uint64 since we have Uint8, Uint16, Uint32...

**Review by dapplion — APPROVED** [review]

**Comment by dapplion** [comment]
Awesome work on the PR thanks for the push on this effort! Looking forward to see all the improvements compounded

**Pull request merged**